---
layout: post
title:  "[ES6+] Promises.finally()"
categories: js 
comments: true

---

<br>

## promise.finally()

~~~javascript
const p1 = new Promise(resolve =>{
  setTimeout(resolve, 5000, "First");
}).then(value => console.log(value))
.finally(()=> console.log("i'm done"));

//5초 뒤 First i'm done
~~~

then을 실행하고 마지막에 finally가 실행된다

에러도 똑같이 작동한다.

에러가 나든 실행이 되든 어쨌거나 finally는 작동한다



<br>

<br>

<br>

<br>

<Br>

노마드코더의 'ES6의 정석'을 듣고 정리 =)











